mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -175,19 +175,22 @@ public void delete(Mailbox mailbox, MailboxMessage message) {
         CompletableFuture<Stream<Pair<CassandraMessageDAO.MessageWithoutAttachment, Stream<CassandraMessageDAO.MessageAttachmentRepresentation>>>>
             messageRepresentations = messageDAO.retrieveMessages(messageIds, fetchType, limit);
         if (fetchType == FetchType.Body || fetchType == FetchType.Full) {
-            return messageRepresentations
-                .thenCompose(stream -> CompletableFutureUtil.allOf(
-                    stream.map(pair -> attachmentLoader.getAttachments(pair.getRight().collect(Guavate.toImmutableList()))
-                        .thenApply(attachments -> Pair.of(pair.getLeft(), attachments)))
-                ))
-                .thenApply(stream -> stream.map(Throwing.function(pair -> pair.getLeft()
-                    .toMailboxMessage(pair.getRight()
-                        .stream()
-                        .collect(Guavate.toImmutableList())))));
+            return CompletableFutureUtil.chainOnStream(messageRepresentations)
+                .thenComposeOnAll(pair ->
+                    attachmentLoader.getAttachments(pair.getRight().collect(Guavate.toImmutableList()))
+                        .thenApply(attachments -> Pair.of(pair.getLeft(), attachments))
+                )
+                .map(Throwing.function(pair ->
+                    pair.getLeft()
+                        .toMailboxMessage(pair.getRight()
+                            .stream()
+                            .collect(Guavate.toImmutableList()))))
+                .completableFuture();
         } else {
-            return messageRepresentations.thenApply(
-                stream -> stream.map(pair ->
-                pair.getLeft().toMailboxMessage(ImmutableList.of())));
+            return CompletableFutureUtil.map(messageRepresentations, pair ->
```

Comment:
For me the fluent is usefull when you need more than one function. Here if use fluent I will have the same code but with .chainOfStream and .futurCompletable(). IMO, it will make the code heavier